gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/TimeAggregationsRequestHandler.java

```diff
@@ -54,63 +55,56 @@ QueryRequest buildQueryRequest(
     builder.setFilter(
         constructQueryServiceFilter(request, requestContext, attributeMetadataProvider));
 
+    List<OrderByExpression> orderByExpressions = getRequestOrderByExpressions(request);
+
     // 3.  Add TimeAggregations
-    addTimeAggregationsToRequest(request, builder, requestContext, attributeMetadataProvider);
+    addTimeAggregationsToRequest(
+        request,
+        builder,
+        requestContext,
+        attributeMetadataProvider,
+        getTimeIntervalOrderByExpression(orderByExpressions));
```

Comment:
Doesn't look like there's a strong reason - we're reusing a converter that works in arrays, but that's easily dealt with either by mapping the order bys rather than filtering, or creating a single arg version of the converter.

Reply:
Done